client/components/domains/register-domain-step/index.jsx

```diff
@@ -1390,7 +1390,9 @@ class RegisterDomainStep extends Component {
 
 		const isSubDomainSuggestion = get( suggestion, 'isSubDomainSuggestion' );
 		if ( ! hasDomainInCart( this.props.cart, domain ) && ! isSubDomainSuggestion ) {
-			this.setState( { pendingCheckSuggestion: suggestion } );
+			// First add the domain
+			this.setState( { pendingCheckSuggestion: null } );
+			this.props.onAddDomain( suggestion, position );
```

Comment:
Does this need gating behind `shouldUseMultipleDomainsInCart`?

Reply:
Yep, good catch!